File: 20221102/모범답안/20221102_01/CandyPrice_test.py
import unittest

from src.com.luckydave.algorithm.fastcampus.easy.CandyPrice.CandyPrice import solution


class CustomTestCase(unittest.TestCase):

    def test_1(self):
        # given
        n = 11
        price = 10

        # when
        result = solution(n, price)

        # then
        self.assertEqual(100, result)

    def test_2(self):
        # given
        n = 90
        price = 38

        # when
        result = solution(n, price)

        # then
        self.assertEqual(3116, result)

    def test_3(self):
        # given
        n = 98
        price = 65

        # when
        result = solution(n, price)

        # then
        self.assertEqual(5850, result)

    def test_4(self):
        # given
        n = 48
        price = 85

        # when
        result = solution(n, price)

        # then
        self.assertEqual(3740, result)

    def test_5(self):
        # given
        n = 23
        price = 26

        # when
        result = solution(n, price)

        # then
        self.assertEqual(546, result)

    def test_6(self):
        # given
        n = 27
        price = 79

        # when
        result = solution(n, price)

        # then
        self.assertEqual(1975, result)


if __name__ == '__main__':
    unittest.main()

File: 20221102/모범답안/20221102_01/CandyPrice_test.js
import assert from 'assert'
import solution from './CandyPrice.js'

describe('Custom Test', function () {
  it('test 1', function (done) {
    // given
    const n = 11
    const price = 10

    // when
    const result = solution(n, price)

    // then
    assert.deepStrictEqual(100, result)
    done();
  });

  it('test 2', function (done) {
    // given
    const n = 90
    const price = 38

    // when
    const result = solution(n, price)

    // then
    assert.deepStrictEqual(3116, result)
    done();
  });

  it('test 3', function (done) {
    // given
    const n = 98
    const price = 65

    // when
    const result = solution(n, price)

    // then
    assert.deepStrictEqual(5850, result)
    done();
  });

  it('test 4', function (done) {
    // given
    const n = 48
    const price = 85

    // when
    const result = solution(n, price)

    // then
    assert.deepStrictEqual(3740, result)
    done();
  });

  it('test 5', function (done) {
    // given
    const n = 23
    const price = 26

    // when
    const result = solution(n, price)

    // then
    assert.deepStrictEqual(546, result)
    done();
  });

  it('test 6', function (done) {
    // given
    const n = 27
    const price = 79

    // when
    const result = solution(n, price)

    // then
    assert.deepStrictEqual(1975, result)
    done();
  });
})
